src/planner/optimalretimer/KinodynamicTimer.cpp

```diff
@@ -0,0 +1,292 @@
+#include "aikido/planner/optimalretimer/KinodynamicTimer.hpp"
+#include <dart/dart.hpp>
+#include <aikido/common/Spline.hpp>
+#include <aikido/common/StepSequence.hpp>
+
+#include "Path.h"
+#include "Trajectory.h"
+
+namespace aikido {
+namespace planner {
+namespace optimalretimer {
+
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline>
+createSplineFromWaypointsAndConstraints(
```

Comment:
In order to match the parabolic smoother implementation more closely, I think it would be helpful to replace this function with three functions in the `detail` namespace:
- `convertToKunzTrajectory(Interpolated traj, VectorXd maxVelocity, VectorXd maxAcceleration, double maxDeviation, double timestep)`
- `convertToKunzTrajectory(Spline traj, VectorXd maxVelocity, VectorXd maxAcceleration, double maxDeviation, double timestep)`
- `convertToSpline(Trajectory traj, double startTime, StateSpace stateSpace)`

Then, `computeOptimalTiming` should basically look like `computeParabolicTiming`.